src/resource.js

```diff
@@ -71,12 +72,19 @@ export default class Resource {
    * @returns {String|Array|Object}
    */
   get source() {
```

Comment:
This function logic seems a little bit complicated - may be it's possible to make it more straightforward using comments and groups? We also should analyze it for vulnerabilities (accessing e.g. `/etc/passwd` using relative paths) so this code should be pretty clear to understand. 

For example:
```javascript
// Inline data
if (this.type === 'inline') return source;

// Local data
...

// Remote data
...
```